src/main/java/io/stargate/sgv2/jsonapi/service/cqldriver/executor/TableSchemaObject.java

```diff
@@ -1,22 +1,98 @@
 package io.stargate.sgv2.jsonapi.service.cqldriver.executor;
 
+import com.datastax.oss.driver.api.core.CqlIdentifier;
+import com.datastax.oss.driver.api.core.data.ByteUtils;
+import com.datastax.oss.driver.api.core.metadata.schema.ColumnMetadata;
+import com.datastax.oss.driver.api.core.metadata.schema.IndexMetadata;
 import com.datastax.oss.driver.api.core.metadata.schema.TableMetadata;
+import com.datastax.oss.driver.api.core.type.VectorType;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.stargate.sgv2.jsonapi.service.schema.SimilarityFunction;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
 
 public class TableSchemaObject extends TableBasedSchemaObject {
 
   public static final SchemaObjectType TYPE = SchemaObjectType.TABLE;
 
-  public TableSchemaObject(TableMetadata tableMetadata) {
+  private final List<VectorConfig> vectorConfigs;
+
+  private TableSchemaObject(TableMetadata tableMetadata, List<VectorConfig> vectorConfigs) {
     super(TYPE, tableMetadata);
+    this.vectorConfigs = vectorConfigs;
   }
 
   @Override
-  public VectorConfig vectorConfig() {
-    return VectorConfig.notEnabledVectorConfig();
+  public List<VectorConfig> vectorConfigs() {
+    return vectorConfigs;
   }
 
   @Override
   public IndexUsage newIndexUsage() {
     return IndexUsage.NO_OP;
   }
+
+  /**
+   * Get table schema object from table metadata
+   *
+   * @param tableMetadata
+   * @param objectMapper
+   * @return
+   */
+  public static TableSchemaObject getTableSettings(
+      TableMetadata tableMetadata, ObjectMapper objectMapper) {
+    Map<String, String> extensions =
+        (Map<String, String>)
+            tableMetadata.getOptions().get(CqlIdentifier.fromInternal("extensions"));
+    String vectorize = extensions != null ? extensions.get("vectorize") : null;
+    Map<String, VectorConfig.VectorizeConfig> resultMap = new HashMap<>();
+    if (vectorize != null) {
+      String vectorizeJson = new String(ByteUtils.fromHexString(vectorize).array());
+      // Define the TypeReference for Map<String, VectorConfig.VectorizeConfig>
+      TypeReference<Map<String, VectorConfig.VectorizeConfig>> typeRef =
+          new TypeReference<Map<String, VectorConfig.VectorizeConfig>>() {};
+
+      // Convert JSON string to Map
+      try {
+        resultMap = objectMapper.readValue(vectorizeJson, typeRef);
+      } catch (JsonProcessingException e) {
+        throw new RuntimeException(e);
```

Comment:
This can occur due to various reasons so should probably use proper exception ("invalid vectorize configuration, root cause: ...") to indicate it is due to somehow invalid stored configuration.

Reply:
One possible improvement idea: first use `ObjectMapper.readTree()` to find invalid JSON (some garbage not JSON), and then checking it's `jsonNode.isObject()`; then traverse key/value pairs and use `mapper.treeToValue()` (to check structure matches). This way could more accurately indicate type of error.